### ISSUE TITLE
Raise exception if both model and potential_fn are provided as arguments to HMC

### ISSUE THREAD

**Issue by jiamings** [pyro-ppl/pyro#1899]
### Issue Description
The potential_fn functionality currently implemented in #1845 does not seem to work as I had expected. Regardless of the potential_fn argument the model seems to use its own potential function?

I could fix this by adding two hacks:
1. If potential_fn is provided, then ignore the one automatically generated from the model (line 234 at hmc.py)
2. Modify the potential_fn to accept {} arguments, somehow at the first step the initial_params = {}.

### Environment
For any bugs, please provide the following:
 - OS and python version: ubuntu, python 3.5
 - PyTorch version: 1.1
 - Pyro version: pyro_ppl-0.3.3+15f3a6d

### Code Snippet
Provide any relevant code snippets and commands run to replicate the issue.

```[python]
    def model():
        return pyro.sample('x', pyro.distributions.Normal(0.0, 1.0))

    def potential_fn(d):
        if 'x' in d.keys():
            std = 0.5
            return d['x'].pow(2).sum() / (2 * std ** 2)
        else:
            return torch.tensor(0.0, requires_grad=True) # d['x'].abs().sum() / 2.0

    hmc_kernel = HMC(model, potential_fn=potential_fn, step_size=0.1, num_steps=4)
    mcmc_run = MCMC(hmc_kernel, num_samples=5000, num_chains=1, warmup_steps=10, disable_progbar=False).run()
    print(hmc_kernel.initial_params)
    print(mcmc_run.marginal())
    samples = mcmc_run.marginal(sites=['x']).empirical['x']._get_samples_and_weights()[0]
    print(samples)
    print(torch.mean(samples), torch.std(samples))
```

**Comment by neerajprad** [comment]
HMC takes either a model or a `potential_fn`. There is no need to provide both (see https://github.com/pyro-ppl/pyro/blob/dev/tests/infer/mcmc/test_hmc.py#L274). The model argument is mostly for backwards compatibility.

On our end, I think we should set `model=None` as default and check that only one of them is provided.

**Comment by jiamings** [comment]
I see, thanks! The test works well.
But the need to define initial_params outside the init function seems a bit hacky to me as well.

I did not see the test before, and it seems that I used the wrong approach to obtain the samples when model=None.

**Comment by neerajprad** [comment]
> But the need to define initial_params outside the init function seems a bit hacky to me as well.

Thanks for the suggestion. I am refactoring this as we speak, and will make it a point to address this in the interface.

**Comment by fehiepsi** [comment]
This issue is addressed in #1913.
